marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -280,6 +283,13 @@ public RunStepResponse run() {
         inputFilePath = (String)fileLocation.get("inputFilePath");
         inputFileType = (String)fileLocation.get("inputFileType");
         outputURIReplacement = (String)fileLocation.get("outputURIReplacement");
+        if(inputFileType.equalsIgnoreCase("csv") && fileLocation.get(separator) != null) {
```

Comment:
The "if" condition has to be 
if(inputFileType.equalsIgnoreCase("csv") && fileLocation.get("separator") != null)

Reply:
https://github.com/marklogic/marklogic-data-hub/blob/574d4b7f2ac4c9dd0bb6f9fb2ccb6bda3d10c0b8/marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java#L286-L291

 line 291 separator cannot be null in the if block. We can check like:
csvColumnSeparator = separator.isNotEmpty ? separator.charAt(0) :  ',';

Reply:
```suggestion
we do not need a member variable "csvColumnSeparator", instead
private char csvColumnSeparator = ','; => private static final CSV_DEFAULT_COLUMN_SEPARATOR = ','; 
       separator = Character.toString(',');
        if (inputFileType.equalsIgnoreCase("csv") && fileLocation.get("separator") != null) {
           String tmp =((String) fileLocation.get("separator")).trim(); 
           if (tmp.equals("\\t") {
               separator = "\t";
           }
        }
        
     https://github.com/marklogic/marklogic-data-hub/blob/574d4b7f2ac4c9dd0bb6f9fb2ccb6bda3d10c0b8/marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java#L627
     
     changed line 627 => .withColumnSeparator(separator.charAt(0));
```